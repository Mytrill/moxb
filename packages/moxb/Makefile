all:
	cd ../.. && $(MAKE)

# this will forward any other make goal to the parent makefile
$(MAKECMDGOALS):
	cd ../.. && $(MAKE) $(MAKECMDGOALS)
