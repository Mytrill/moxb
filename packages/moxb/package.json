{
  "name": "@moxb/moxb",
  "version": "0.1.3",
  "description": "MobX data binding",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "main": "build",
  "typings": "build/",
  "files": [
    "build/**/*"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/moxb/moxb.git"
  },
  "author": "moxb (moxb.org)",
  "license": "MIT",
  "dependencies": {
    "tslib": "^1.9.3"
  },
  "peerDependencies": {
    "mobx": "^4.3.0 || ^5.0.0"
  },
  "devDependencies": {
    "@types/jest": "^23.1.5",
    "@types/meteor": "^1.4.16",
    "jest": "^23.3.0",
    "prettier": "^1.13.7",
    "ts-jest": "^23.0.0",
    "typescript": "^2.9.2"
  },
  "bugs": {
    "url": "https://github.com/moxb/moxb/issues"
  },
  "homepage": "https://github.com/moxb/moxb#readme",
  "keywords": [
    "mobx",
    "UI"
  ]
}
