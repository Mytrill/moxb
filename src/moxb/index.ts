export * from './impl';
export * from './meteor';
export * from './types';
