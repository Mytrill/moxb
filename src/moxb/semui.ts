export * from './action/ActionUi';
export * from './bind/BindUi';
export * from './bool/BoolUi';
export * from './confirm/ConfirmUi';
export * from './many-of/ManyOfUi';
export * from './modal/ModalUi';
export * from './numeric/NumericUi';
export * from './one-of/OneOfUi';
export * from './table/TablePaginationUi';
export * from './table/TableSearchUi';
export * from './table/TableUi';
export * from './text/TextUi';
