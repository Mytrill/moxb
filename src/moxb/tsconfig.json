{
    "compilerOptions": {
        "module": "commonjs",
        "target": "ES5",
        "jsx": "react",
        "lib": ["DOM","ES6","ES2015","ES2017", "DOM.Iterable","ScriptHost"],
        "importHelpers":true,
        "experimentalDecorators": true,
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strict": true,
        "outDir": "../build",
        "declaration": true,
        "skipLibCheck": true,
        "strictNullChecks": true,
        "removeComments": false
    },
    "compilerOptions": {
        "baseUrl": "",
        "paths": {
            "*": [
                "*",
                "../../optional-modules/node_modules/*"
            ]
        }
    },
    "exclude": [
        "../node_modules"
    ]
}
