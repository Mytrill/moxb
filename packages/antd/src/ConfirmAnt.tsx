import { observer } from 'mobx-react';
import * as React from 'react';
import { Modal } from 'antd';
import { ModalFuncProps } from 'antd/lib/modal';
import { Confirm as MoxbConfirm } from '@moxb/moxb';
import { parseProps } from './BindAnt';
import { BindMarkdownDiv } from './LabelAnt';

export interface BindConfirmAntProps extends ModalFuncProps {
    operation: MoxbConfirm<any>;
    invisible?: boolean;
}

@observer
export class ConfirmAnt extends React.Component<BindConfirmAntProps> {
    render() {
        const { operation, ...props } = parseProps(this.props, this.props.operation);
        return (
            <Modal
                {...props as any}
                onOk={operation.onConfirm}
                visible={operation.open}
                onCancel={operation.onCancel}
                cancelText={operation.cancelButton.label}
                okText={operation.confirmButton.label}
                title={operation.header}
            >
                <BindMarkdownDiv text={operation.content} />
            </Modal>
        );
    }
}
