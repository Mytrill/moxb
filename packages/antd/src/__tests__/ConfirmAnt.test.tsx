import { ConfirmImpl, BindImpl } from '@moxb/moxb';
import { shallow } from 'enzyme';
import * as React from 'react';
import { ConfirmAnt } from '../ConfirmAnt';

describe('ConfirmAnt', function() {
    it('should render a confirm dialog by default', function() {
        const operation = new ConfirmImpl<any>({
            cancelButton: new BindImpl({
                id: 'ApplicationImpl.noConfirm',
                label: 'Cancel',
            }),
            confirmButton: new BindImpl({
                id: 'ApplicationImpl.yesConfirm',
                label: 'Do the action',
            }),
            content: () => 'Do you really want to execute the action?',
            header: () => 'Confirm dialog',
            confirm: () => alert('You confirmed the action'),
        });
        expect(shallow(<ConfirmAnt operation={operation} />)).toMatchSnapshot();
    });
});
