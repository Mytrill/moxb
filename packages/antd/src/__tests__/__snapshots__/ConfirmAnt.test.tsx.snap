// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmAnt should render a confirm dialog by default 1`] = `
<Modal
  cancelButtonDisabled={false}
  cancelText="Cancel"
  confirmLoading={false}
  maskTransitionName="fade"
  okButtonDisabled={false}
  okText="Do the action"
  okType="primary"
  onCancel={[Function]}
  onOk={[Function]}
  prefixCls="ant-modal"
  title={
    <BindMarkdownDiv
      text=""
    />
  }
  transitionName="zoom"
  visible={false}
  width={520}
>
  <BindMarkdownDiv
    text=""
  />
</Modal>
`;
